plugins/woocommerce-admin/client/tasks/fills/shipping/index.js

```diff
@@ -328,12 +334,153 @@ export class Shipping extends Component {
+								this.completeStep();
+							} }
+							createNotice={ createNotice }
+						/>
+					),
+				},
+				label_printing: {
+					label: __(
+						'Enable shipping label printing and discounted rates',
+						'woocommerce'
+					),
+					description: __(
+						'Save time and fulfill your orders with WooCommerce Shipping. You can manage it at any time in WooCommerce Shipping Settings.',
+						'woocommerce'
+					),
+					content: (
+						<>
+							<WCSBanner />
+							<Plugins
+								onComplete={ ( plugins, response ) => {
+									createNoticesFromResponse( response );
+									recordEvent(
+										'tasklist_shipping_label_printing',
+										{
+											install: true,
+											plugins_to_activate:
+												pluginsToActivate,
+										}
+									);
+									this.completeStep();
+								} }
+								onError={ ( errors, response ) =>
+									createNoticesFromResponse( response )
+								}
+								onSkip={ () => {
+									recordEvent(
+										'tasklist_shipping_label_printing',
+										{
+											install: false,
+											plugins_to_activate:
+												pluginsToActivate,
+										}
+									);
+									getHistory().push(
+										getNewPath( {}, '/', {} )
+									);
+									onComplete();
+								} }
+								pluginSlugs={ pluginsToActivate }
+							/>
+							{ ! isJetpackConnected && (
+								<Text
+									variant="caption"
+									className="woocommerce-task__caption"
+									size="12"
+									lineHeight="16px"
+									style={ { display: 'block' } }
+								>
+									{ interpolateComponents( {
+										mixedString: agreementText,
+										components: {
+											link: (
+												<Link
+													href={
+														'https://wordpress.com/tos/'
+													}
+													target="_blank"
+													type="external"
+												>
+													<></>
+												</Link>
+											),
+										},
+									} ) }
+								</Text>
+							) }
+						</>
+					),
+				},
+				store_location: {
+					label: __( 'Set your store location', 'woocommerce' ),
+					description: __(
+						'Add your store location to help us calculate shipping rates and the best shipping options for you. You can manage your store location again at any time in WooCommerce Settings General.',
+						'woocommerce'
+					),
+					buttonText: __( 'Save store location', 'woocommerce' ),
+				},
+			};
+
+			for ( const i in shippingSmartDefaultsSteps ) {
+				for ( const j in steps ) {
+					if ( steps[ j ].key === i ) {
+						steps[ j ] = {
+							...steps[ j ],
+							...shippingSmartDefaultsSteps[ i ],
+						};
+					}
+				}
+			}
+		}
+
 		return filter( steps, ( step ) => step.visible );
 	}
 
 	render() {
 		const { isPending, step } = this.state;
 		const { isUpdateSettingsRequesting } = this.props;
+		const steps = this.getSteps();
+		// Empty description field if it's not the current step.
+		if ( this.shippingSmartDefaultsEnabled ) {
+			for ( const i in steps ) {
```

Comment:
How about moving this logic to `getSteps` directly?

Reply:
Good suggestion! I'll before merging the PR 👍